src/GraphQL.Tests/Bugs/Bug1205VeryLongInt.cs

```diff
@@ -1,17 +1,56 @@
 using GraphQL.Types;
+using Shouldly;
+using System;
 using Xunit;
 
 namespace GraphQL.Tests.Bugs
 {
     // https://github.com/graphql-dotnet/graphql-dotnet/issues/1205
+    // https://github.com/graphql-dotnet/graphql-dotnet/issues/1022
     public class Bug1205VeryLongInt : QueryTestBase<Bug1205VeryLongIntSchema>
     {
         [Fact]
-        public void Very_Long_Int_Should_Return_As_Is()
+        public void Very_Long_Number_Should_Return_Error_For_Int()
         {
-            var query = "{ big }";
+            var query = "{ int }";
+            var expected = new ExecutionResult { Errors = new ExecutionErrors { new ExecutionError("Value was either too large or too small for an Int32.", new OverflowException()) } };
+            AssertQueryIgnoreErrors(query, expected, renderErrors: true, expectedErrorCount: 1);
+        }
+
+        [Fact]
+        public void Very_Long_Number_In_Input_Should_Return_Error_For_Int()
```

Comment:
should the "bigger than a long scenario" also be tested?

Reply:
I agree. I also thought that we really lack the `BigInteger` scalar. With its appearance we will have a complete picture of scalars for integer types. I have added tests.